src/Hosting/Hosting/src/Internal/Host.cs

```diff
@@ -17,16 +18,18 @@ internal class Host : IHost
       , IAsyncDisposable
 #endif
     {
+        private readonly IConfiguration _configuration;
         private readonly ILogger<Host> _logger;
         private readonly IHostLifetime _hostLifetime;
         private readonly ApplicationLifetime _applicationLifetime;
         private readonly HostOptions _options;
         private IEnumerable<IHostedService> _hostedServices;
 
-        public Host(IServiceProvider services, IHostApplicationLifetime applicationLifetime, ILogger<Host> logger,
-            IHostLifetime hostLifetime, IOptions<HostOptions> options)
+        public Host(IServiceProvider services, IConfiguration configuration, IHostApplicationLifetime applicationLifetime,
```

Comment:
It smells that we'd need to dispose a service that was provided to us by DI. I think we can get the container to dispose it for us.
https://github.com/aspnet/Extensions/blob/b28d96763fdf2750f31936641313f3af24def0fc/src/Hosting/Hosting/src/HostBuilder.cs#L200
When registering the config, if you register it as a factory that returns the instance then the container should dispose the instance when it's done.

Reply:
I offered that before but @davidfowl [chose to inject it explicitly instead](https://github.com/aspnet/Extensions/issues/938#issuecomment-456709911).

Notet hat the `Host` also disposes the service provider (which is also provided by DI), so it’s already doing that in a way.

Reply:
(But I still do think that disposing it implicitly through the service provider would be a lot cleaner. Since the configuration is a singleton, the factory action would only be called once, right? So this wouldn’t actually have much of a drawback except for being a bit cryptic.)

Reply:
Ah, I'll talk to him.

Reply:
I talked to @davidfowl offline, let's try the factory approach. I don't think he read your original proposal all the way through.

Reply:
Alright, that works for me, thanks for clearing that up! (Note to self: Try to write shorter issues 😅)

Reply:
Make sure you include a comment that the factory is used as a means of requesting disposal.

Reply:
Updated the PR. Had to keep the dependency on `IConfiguration` within `Host` though to make sure that there’s something that resolves the configuration. Otherwise it wouldn’t have been picked up reliably by the service provider.

Reply:
How about resolving IConfiguration here instead?
https://github.com/aspnet/Extensions/blob/b28d96763fdf2750f31936641313f3af24def0fc/src/Hosting/Hosting/src/HostBuilder.cs#L222-L227

Reply:
Works for me – good idea to avoid having to rely on the host for that! 👍